### PR TITLE
fix audit diff in kz_services

### PR DESCRIPTION
**Pull request by icehess**
`kz_json:diff/2` is using `sets:subtract/2` which means it has all elements of
current quantities if they are not in proposed too. This means if something new
is added to proposed which was not previously in current, the diff would be
empty.

To fix this we calculate the subtract of both current - proposed and proposed
- current, so we have all old/new values, make current quatities negative and
do a sum to get the proper diff.

### CODE REVIEW

**Comment by icehess** [comment]
For example, assume we have these quantities:

```erlang
    Current = kz_json:from_list_recursive(
                [{<<"quantities">>
                 ,[{<<"account">>
                   ,[{<<"branding">>, [{<<"whitelabel">>,1}]}
                    ,{<<"accounts">>, [{<<"account">>, 1}]}
                    ,{<<"devices">>, [{<<"sip_device">>, 1}]}
                    ,{<<"users">>, [{<<"admin">>, 1}]}
                    ,{<<"ports">>, [{<<"submitted">>, 2}]}
                    ,{<<"lentils">>, [{<<"soup">>, 1}]}
                    ]
                   }
                  ,{<<"cascade">>, {[]}}
                  ,{<<"manual">>, [{<<"support">>, [{<<"support">>, 123}]}]}
                  ]
                 }
                ]).
    Propose = kz_json:from_list_recursive(
                [{<<"quantities">>
                 ,[{<<"account">>
                   ,[{<<"branding">>, [{<<"whitelabel">>,1}]}
                    ,{<<"accounts">>, [{<<"account">>, 1}]}
                    ,{<<"devices">>, [{<<"sip_device">>, 1}]}
                    ,{<<"users">>, [{<<"admin">>, 2}]}
                    ,{<<"ports">>, [{<<"submitted">>, 1}]}
                    ,{<<"dummy">>, [{<<"dumper">>, 1}]}
                    ]
                   }
                  ,{<<"cascade">>, [{<<"accounts">>, [{<<"account">>, 1}]}]}
                  ,{<<"manual">>, {[]}}
                  ]
                 }
                ]).
```

Old `generate_diff_obj` would crash on:

```erlang
** exception error: no match of right hand side value [<<"cascade">>]
     in function  kz_services:calculate_diffs/2 (src/kz_services.erl, line 1269)
```

If we remove cascade from current and run it again:

```erlang
[{[{<<"category">>,<<"support">>},
   {<<"source">>,<<"support">>},
   {<<"quantity">>,-123},
   {<<"billable">>,0}]},
 {[{<<"category">>,<<"ports">>},
   {<<"source">>,<<"submitted">>},
   {<<"quantity">>,-1},
   {<<"billable">>,1}]},
 {[{<<"category">>,<<"lentils">>},
   {<<"source">>,<<"soup">>},
   {<<"quantity">>,-1},
   {<<"billable">>,0}]},
 {[{<<"category">>,<<"users">>},
   {<<"source">>,<<"admin">>},
   {<<"quantity">>,1},
   {<<"billable">>,2}]}]
```

Notice that cascade account and dummy items from proposed service is missing.

New run it again with the fixed one would return:

```erlang
[{[{<<"category">>,<<"users">>},
   {<<"source">>,<<"admin">>},
   {<<"quantity">>,1},
   {<<"billable">>,2}]},
 {[{<<"category">>,<<"lentils">>},
   {<<"source">>,<<"soup">>},
   {<<"quantity">>,-1},
   {<<"billable">>,0}]},
 {[{<<"category">>,<<"support">>},
   {<<"source">>,<<"support">>},
   {<<"quantity">>,-123},
   {<<"billable">>,0}]},
 {[{<<"category">>,<<"ports">>},
   {<<"source">>,<<"submitted">>},
   {<<"quantity">>,-1},
   {<<"billable">>,1}]},
 {[{<<"category">>,<<"dummy">>},
   {<<"source">>,<<"dumper">>},
   {<<"quantity">>,1},
   {<<"billable">>,1}]},
 {[{<<"category">>,<<"accounts">>},
   {<<"source">>,<<"account">>},
   {<<"quantity">>,1},
   {<<"billable">>,1}]}]
```

**Comment by jamesaimonetti** [comment]
Shouldn't your example above be a test to demonstrate the point?

File: kazoo_services/src/kz_services.erl [hunk 1]
```diff
@@ -1195,18 +1195,18 @@ save_services_jobj(Services, ProposedJObj) ->
...
+            audit_reseller(Services, Diff)
     end.
 
 %%------------------------------------------------------------------------------
 %% @doc
 %% @end
 %%------------------------------------------------------------------------------
--spec audit_reseller(services(), kz_json:object(), kz_json:object()) -> 'ok'.
-audit_reseller(Services, CurrentServicesJObj, ProposedServicesJObj) ->
+-spec audit_reseller(services(), kz_json:object()) -> 'ok'.
```
Comment (jamesaimonetti) [inline kazoo_services/src/kz_services.erl:1208]: Should be `kz_json:objects()` for the Diff, right?

**Comment by jamesaimonetti** [comment]
Merged, not sure why Github isn't recognizing that...

**Pull request closed without merging**